src/core/server/http/http_service.mock.ts

```diff
@@ -30,6 +32,8 @@ const createSetupContractMock = () => {
     setBasePathFor: jest.fn(),
     // we can mock some hapi server method when we need it
     server: {} as Server,
+    createNewServer: async (cfg: Partial<HttpConfig>): Promise<HttpServerSetup> =>
```

Comment:
**nit**: If original function returns a promise, mock function also should return a promise
```typescript
createNewServer: (cfg: Partial<HttpConfig>) => (Promise.resolve({} as HttpServerSetup)),
```